release/release.go

```diff
@@ -105,3 +105,14 @@ type GcpImage struct {
 	Family  string `json:"family,omitempty"`
 	Name    string `json:"name,omitempty"`
 }
+
+// PlatformAzure represents an Azure cloud image
+type PlatformAzure struct {
+	PlatformBase
+	Image *AzureImage `json:"image"`
+}
+
+// AzureImage represents an Azure cloud image
+type AzureImage struct {
```

Comment:
Should we have some kind of digest here?

Reply:
Hmm yes...very good point.

I think that the uploaded VHD is in fact the same as the `.vhd.gz`, just uncompressed, but I'm verifying that now.  The value here of having it in the Azure infrastructure to start is to avoid ingress/egress costs I'm sure.  Now on that topic...I think most users are just going to directly ask Azure to create a copy of it; they're not going to download it and verify a sha256sum and reupload.  

But we probably should support that just in case - at least make it easy to verify the uploaded images out of band.

I'm looking at basically just reusing the `Artifact` type for this.

Reply:
I verified that indeed the current Azure VHD is just uncompressed, that's it:

```
[root@bash tmp]# curl -LO https://rhcos.blob.core.windows.net/imagebucket/rhcos-47.83.202012030221-0-azure.x86_64.vhd
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 16.0G  100 16.0G    0     0  31.7M      0  0:08:36  0:08:36 --:--:-- 33.2M
[root@bash tmp]# sha256sum rhcos-47.83.202012030221-0-azure.x86_64.vhd 
6201d201935e9e1cc144ae77d151ea736dd601ecbad246edc9cc1d75a1d93e41  rhcos-47.83.202012030221-0-azure.x86_64.vhd
[root@bash tmp]# 
```

Which matches the uncompressed checksum of [this 4.7 RHCOS build](https://releases-art-rhcos.svc.ci.openshift.org/art/storage/releases/rhcos-4.7/47.83.202012030221-0/x86_64/meta.json).